### DISCUSSION TITLE
Looking for examples of tty / using subprocesses correctly in python

### DISCUSSION THREAD

**Discussion by gwenwindflower** [pomdtr/sunbeam#97] [Q&A] [answered]
attempting to run a terminal command and ideally get an interactive/streaming response back, which my intuition says should be a 'tty' type command, but can't find any examples of the expected type to return in the JSON. 

in the interim trying to do it with 'detail' type,

```python
elif payload["command"] == "test_node":

    def dbt_test(node_name):
        test = subprocess.run(
            ["dbt", "test", "-s", node_name], shell=True, capture_output=True
        )
        return test.stdout

    node_name = payload["params"]["node_name"]
    test_result = dbt_test(node_name)
    print(json.dumps({"format": "markdown", "text": test_result.decode("utf-8")}))
```
this works but i can't figure out how to introspect it and work from here. any tips on using the tty type? am i right about what it's for?

**Comment by gwenwindflower** [comment]
Happy to add some docs/an example too once I get it, I do Developer Experience work for a living, so I write a lot of docs and such.

**Comment by pomdtr** [comment]
The tty allows you to run interactive terminal apps in a sunbeam extension (ex: an editor).

It allows your script to gain control of the terminal (stdin/stdout and stderr are interactive).

Example usage in python would be:

```python
        test = subprocess.run(
            ["dbt", "test", "-s", node_name], stdin=subprocess.Pipe, stdout=subprocess.Pipe, stderr=subprocess.Pipe
        )
```

**Reply by pomdtr** [reply]
However I'm not sure what the purpose of the `dbt test -s` command is. Is it an interactive command ?

**Reply by gwenwindflower** [reply]
do i return anything specific?

**Reply by gwenwindflower** [reply]
it is not actually, but it streams logs of what it's doing and i was hoping to show those in real time rather than pause and then load it all

**Reply by gwenwindflower** [reply]
also ty for the response and help!

**Comment by pomdtr** [comment, marked as answer]
I see. You will be able to see the logs, but they will be hidden as soon as the command exit. 

I'm still searching for a good way to handle this usecase.

**Reply by pomdtr** [reply]
this issue is related: https://github.com/pomdtr/sunbeam/issues/94
